pkg/logql/engine.go

```diff
@@ -214,8 +214,17 @@ func (q *query) resultLength(res promql_parser.Value) int {
 func (q *query) Exec(ctx context.Context) (logqlmodel.Result, error) {
 	sp, ctx := opentracing.StartSpanFromContext(ctx, "query.Exec")
 	defer sp.Finish()
-	log := spanlogger.FromContext(ctx)
-	defer log.Finish()
+	spLogger := spanlogger.FromContext(ctx)
+	defer spLogger.Finish()
+
+	sp.LogKV(
+		"type", GetRangeType(q.params),
+		"query", q.params.Query(),
+		"start", q.params.Start(),
+		"end", q.params.End(),
+		"step", q.params.Step(),
+		"length", q.params.End().Sub(q.params.Start()),
+	)
```

Comment:
`logKV` has this weird pattern where the first key-value pair is actually `event` and the event name.

One more thing: before you merge this, could you test it with a split query in a running cluster? This will work well if there's a span on top of every `Exec`, but if it is a single parent span for all `query.Exec` executions then you'd be emitting the same event multiple times for the same span; in such case, logging the line as you were doing previously is actually better (i.e: my suggestion was crap :rofl: )

```suggestion
        sp, ctx := opentracing.StartSpanFromContext(ctx, "query.Exec")
        defer sp.Finish()
        if sp != nil {
            sp.LogKV(
                "event", "started query exec",
		"type", GetRangeType(q.params),
		"query", q.params.Query(),
		"start", q.params.Start(),
		"end", q.params.End(),
		"step", q.params.Step(),
		"length", q.params.End().Sub(q.params.Start()),
	    )
	    defer sp.LogKV(
                "event", "finished query exec",
		"type", GetRangeType(q.params),
		"query", q.params.Query(),
		"start", q.params.Start(),
		"end", q.params.End(),
		"step", q.params.Step(),
		"length", q.params.End().Sub(q.params.Start()),
	    )
        }
	spLogger := spanlogger.FromContext(ctx)
	defer spLogger.Finish()

	
```

Reply:
Looks like it's working fine ([source][1]).

![image](https://user-images.githubusercontent.com/8354290/235646741-afbb75d9-ace0-43d4-b750-5cb17736d933.png)

> logKV has this weird pattern where the first key-value pair is actually event and the event name.

IIUC, this is just a convention, right? In this particular case, I find it kind of unnecessary as the same info gets printed twice plus we already record an event with the query stats once the query is finished executing.

Here's the working version w/o the finish event ([source][2]):
![image](https://user-images.githubusercontent.com/8354290/235648998-2eb8312d-24a9-4cc3-8119-9599247542ba.png)



[1]: https://admin-dev-us-central-0.grafana.net/grafana/explore?left=%7B%22datasource%22:%22tempo-dev-01%22,%22queries%22:%5B%7B%22query%22:%2239f8ec2fe93e11a1%22,%22queryType%22:%22traceql%22,%22refId%22:%22A%22,%22limit%22:20%7D%5D,%22range%22:%7B%22from%22:%22now-5m%22,%22to%22:%22now%22%7D,%22panelsState%22:%7B%22trace%22:%7B%22spanId%22:%22623b15fbdbb46b74%22%7D%7D%7D&orgId=1
[2]: https://admin-dev-us-central-0.grafana.net/grafana/explore?left=%7B%22datasource%22:%22tempo-dev-01%22,%22queries%22:%5B%7B%22query%22:%225ac20ab385a80918%22,%22queryType%22:%22traceql%22,%22refId%22:%22A%22,%22limit%22:20%7D%5D,%22range%22:%7B%22from%22:%22now-5m%22,%22to%22:%22now%22%7D,%22panelsState%22:%7B%22trace%22:%7B%22spanId%22:%222b2fcbbed16c4584%22%7D%7D%7D&orgId=1

Reply:
> IIUC, this is just a convention, right? In this particular case, I find it kind of unnecessary as the same info gets printed twice plus we already record an event with the query stats once the query is finished executing.

good point, but I think I read somewhere that these different trace libs (OTEL, tempo, jaeger, etc) somehow rely on this pattern. Could you try one more time without the `event` and check how it behaves?

Reply:
> Could you try one more time without the event and check how it behaves?

Looks to be working fine.

![image](https://user-images.githubusercontent.com/8354290/235693686-3d0a4529-5a92-402e-aa8b-025f0aea8895.png)